runtime/codert_vm/cnathelp.cpp

```diff
@@ -682,37 +682,194 @@ old_fast_jitNewObjectNoZeroInit(J9VMThread *currentThread)
 void* J9FASTCALL
 old_slow_jitGetFlattenableField(J9VMThread *currentThread)
 {
-	return NULL;
+	SLOW_JIT_HELPER_PROLOGUE();
+	J9RAMFieldRef* cpEntry = (J9RAMFieldRef*) currentThread->floatTemp1;
+	j9object_t receiver = (j9object_t) currentThread->floatTemp2;
+	j9object_t returnObject = NULL;
+	void *rc = NULL;
+	void *oldPC = currentThread->jitReturnAddress;
+
+	if (NULL == receiver) {
+		buildJITResolveFrameForTrapHandler(currentThread);
+		rc = (void*)old_slow_jitThrowNullPointerException;
+		goto done;
+	}
+
+	buildJITResolveFrameWithPC(currentThread, J9_STACK_FLAGS_JIT_ALLOCATION_RESOLVE | J9_SSF_JIT_RESOLVE, parmCount, true, 0, oldPC);
+	returnObject = currentThread->javaVM->internalVMFunctions->getFlattenableField(currentThread, cpEntry, receiver, FALSE, FALSE);
+	if (NULL == returnObject) {
+		rc = setHeapOutOfMemoryErrorFromJIT(currentThread);
+		goto done;
+	}
+
+	currentThread->floatTemp1 = (void*)returnObject; // in case of decompile
+	rc = restoreJITResolveFrame(currentThread, oldPC, false, false);
+	if (NULL != rc) {
+		goto done;
+	}
+
+	JIT_RETURN_UDATA(returnObject);
+
+done:
+	SLOW_JIT_HELPER_EPILOGUE();
+	return rc;
 }
 
 void* J9FASTCALL
 old_fast_jitGetFlattenableField(J9VMThread *currentThread)
 {
-	return NULL;
+	OLD_JIT_HELPER_PROLOGUE(2);
+	DECLARE_JIT_PARM(J9RAMFieldRef*, cpEntry, 1);
+	DECLARE_JIT_PARM(j9object_t, receiver, 2);
+	j9object_t returnObject = NULL;
+	void *rc = NULL;
+
+	if (NULL == receiver) {
+		buildJITResolveFrameForTrapHandler(currentThread);
```

Comment:
This is the wrong kind of frame build - please follow the other examples in this file, don't try to be too clever re-using the NPE entrypoint.